cuqi/implicitprior/_restorator.py

```diff
@@ -0,0 +1,218 @@
+#%%
+from abc import ABC, abstractmethod
+from cuqi.distribution import Distribution
+import numpy as np
+    
+class RestorationPrior(Distribution):
+    """    
+    This class defines an implicit distribution associated with a restoration operator
+    (eg denoiser). They are several works relating restorations operators with
+    priors, see 
+        -Laumont et al. https://arxiv.org/abs/2103.04715
+        -Hu et al. https://openreview.net/pdf?id=x7d1qXEn1e
+    We cannot sample from this distribution, neither compute its logpdf. It allows
+    us to apply algorithms such as MYULA and PnPULA.
+    
+    Parameters
+    ---------- 
+    restorator : callable f(x, restoration_strength)
+        Function f that accepts input x to be restored and returns the
+        restored version of x and information about the restoration operation.
+        
+    restorator_strength : float
+        Strength of the restoration operation. In the case where the restorator is
+        a denoiser, this parameter might correspond to the noise level.
+        
+    restorator_kwargs : dictionary
+        Dictionary containing information about the restorator.
+    
+    potential : callable function
+        In the case, where the restoration operator is the proximal operator of
```

Comment:
- Should these two sentences be one?   
- "Function potential" is not defined.  
- The last part "this function is a mapping" is very vague, please state more precisely what this mapping is, in addition to where it maps from and to.

Reply:
Good point. We completely agree that it was not very clear. We changed a bit and added a concrete example (TV) to illustrate what we mean by potential.